src/gui/QvisExpressionsWindow.C

```diff
@@ -627,8 +627,10 @@ QvisExpressionsWindow::CreateWindowContents()
 
     connect(exprListBox, SIGNAL(itemSelectionChanged()),
             this, SLOT(UpdateWindowSingleItem()));
-    connect(nameEdit, SIGNAL(textChanged(const QString&)),
+    connect(nameEdit, SIGNAL(textEdited(const QString&)),
```

Comment:
In your testing, did you observe whether or not the nameTextChanged slot was called each time a character was entered in the box? One of the LineEdit signals behaves this way, (I cannot recall exactly which one offhand).  This can lead to bad behavior, or simply more work than is needed. If that's the case, then perhaps just the 'editingFinished' signal should be observed and the nameTextChanged slot used and modified to call 'UpdateExpressionBox' at the end?

Reply:
Sorry, I was looking at an earlier commit, not your final.